### PR TITLE
Added subscription tests and unsubscribe function

### PR DESCRIPTION
**Pull request by trickstival**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -89,9 +89,11 @@ export function buildStore<
     })
   }
 
-  function subscribe(callback: SubscriptionCallback<S>): void {
+  function subscribe(callback: SubscriptionCallback<S>) {
     subscriptions.push(callback)
-    // TODO: return function to remove subscription
+    return () => {
+      subscriptions.splice(subscriptions.indexOf(callback), 1)
```
Comment (posva) [inline src/store.ts]: we need to make sure that the value is > -1. Otherwise `splice` will remove the last element

**Comment by trickstival** [comment]
@posva Done

**Comment by posva** [comment]
What is the _subscriptions for? I would rather not directly expose them

Eduardo San Martin Morote

**Comment by trickstival** [comment]
> What is the _subscriptions for? I would rather not directly expose them Eduardo San Martin Morote
> […](#)
> On 22 Dec 2019, at 15:33, Patrick ***@***.***> wrote: ﻿ @posva Done — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

You're right. At first glance it was the way I found to test if the subscriptions were right. Now I changed it to use spies instead.

**Pull request merged**